EIPS/eip-2767.md

```diff
@@ -0,0 +1,239 @@
+- Governor is the board member who has voting right.
+- Power represents the voting privilege of a governor.
+
+A contract that is compliant with `ERC-2767` shall implement the following interface:
+
+```solidity
+/// @title ERC-2767 Governance with Privileged Voting Rights
+/// @dev ERC-165 InterfaceID: 0x4fe54581
+interface ERC2767 {
+    /// @dev This emits when governor power changes
+    event GovernorPowerUpdated(address indexed governor, uint256 power);
+
+    /// @notice Gets the consensus power of the governor
+    /// @param _governor Address of the governor
+    /// @return The governor's voting power
+    function powerOf(address _governor) external view returns (uint256);
+
+    /// @notice Gets the sum of the power of all governors
+    /// @return Sum of the power of all governors
+    function totalPower() external view returns (uint256);
+
+    /// @notice Gets number votes required for achieving consensus
+    /// @return Required number of consensus votes
+    function required() external view returns (uint256);
+
+    /// @notice Updates governor statuses
+    /// @param _governor Governor address
+    /// @param _newPower New power for the governor
+    /// @dev Only Governance can call
+    function setGovernor(address _governor, uint256 _newPower) external;
+}
+```
+
+Implementations that require equal governance rights between the governors can simply have equal power values for every governor.
+
+There is a possibility that `totalPower()` changes rapidly due to adding or removing governors. This can result in having a fixed `required()` return value easier or difficult to achieve. Implementations can have custom logic that affect the return value of `required()`. For example the return value of `required()` can be changed by custom logic in order to maintain `51%` of `totalPower()`. So if more governors are added in such implementations, the `required()` return value would automatically take that into account.
+
+### Interface Identification
+
+An `ERC-2767` Governance Contract should also implement `ERC-165`. This helps general tools to identify whether a contract is a `ERC-2767` Governance contract.
+
+```solidity
+interface ERC165 {
+    /// @notice Query if a contract implements an interface
+    /// @param interfaceID The interface identifier, as specified in ERC-165
+    /// @dev Interface identification is specified in ERC-165. This function
+    ///  uses less than 30,000 gas.
+    /// @return `true` if the contract implements `interfaceID` and
+    ///  `interfaceID` is not 0xffffffff, `false` otherwise
+    function supportsInterface(bytes4 interfaceID) external view returns (bool);
+}
+```
+
+### Transaction execution (Optional)
+
+There are two optional interfaces for transaction execution: On-chain and Off-chain. A `ERC-2767` compatible governance contract can optionally implement one of them or both. The contract should specify the interfaceID(s) using `ERC-165`'s `supportsInterface` method.
```

Comment:
This whole section feels out of place in a standard.  I see very little value in standardizing how contracts are called by their owners.  Each owner can choose to use different tools/mechanisms and that is OK.  Recommend including a section in the rationale why you think this is something that should be standardized.

If you do think standardization of contract execution is valuable, then I think it should be a separate EIP.  As far as standards go, many short standards tend to be better than one monolithic standard.  It lets people pick and choose what they want to implement.

Reply:
Yes, agree there is some merit in not standardizing how contracts are called by their owners.  But there is also some usefulness to it.  That's why these voting interfaces are optional.

1. Standardizing the voting makes it easier to make user interfaces and tools without contract integration problems.
2. Standardizing the voting makes it easier to understand and adopt implementations because they follow the standard.
3. Every choice a standard makes is a choice that implementors don't have to figure out and choose which makes the implementor's job easier as long as the standard is flexible and correct enough not to get in the way.